plugins/carbon_black_defense/komand_carbon_black_defense/connection/connection.py

```diff
@@ -1,35 +1,119 @@
         self.host = params.get(Input.URL)
         self.token = params.get(Input.API_KEY).get("secretKey")
+        self.org_key = params.get(Input.ORG_KEY)
         self.connector = params.get(Input.CONNECTOR)
+        self.headers = {"X-Auth-Token": f"{self.token}/{self.connector}"}
+
+    def get_job_id_for_enriched_event(self, process_name: str, event_id: str) -> Optional[str]:
+        response = self.call_api(
+            "POST",
+            f"{self.host}/api/investigate/v2/orgs/{self.org_key}/enriched_events/search_jobs",
+            json_data={
+                "criteria": {
+                    "process_name": [process_name],
+                },
+                "event_ids": event_id,
+            },
+        ).get("job_id")
+
+        self.logger.info(f"The response is {response}.")
+        self.logger.info(f"The url is: {self.host}/api/investigate/v2/orgs/{self.org_key}/enriched_events/search_jobs)")
+        if response:
+            job_id = response
+            return job_id
+        return None
 
-    def test(self):
-        host = self.host
-        token = self.token
-        connector = self.connector
-        devices = "/integrationServices/v3/device"
-        headers = {"X-Auth-Token": f"{token}/{connector}"}
-        url = host + devices
-
-        result = requests.get(url, headers=headers)
-        if result.status_code == 200:
-            return {"success": True}
-        if result.status_code == 401:
-            raise ConnectionTestException(preset=ConnectionTestException.Preset.API_KEY)
-        raise ConnectionTestException(
-            f"An unknown error occurred. Response code was: {result.status_code}"
-            f" If the problem persists please contact support for help. Response was: {result.text}"
+    def get_enriched_event_status(self, job_id: str = None):
+        response = self.call_api(
+            "GET",
+            f"{self.host}/api/investigate/v1/orgs/{self.org_key}/enriched_events/search_jobs/{job_id}",
+            json_data={"cb_job_id": job_id},
         )
+        contacted = response.get("contacted")
+        completed = response.get("completed")
+        if contacted and completed and (contacted == completed):
+            return True
+        return False
+
+    def retrieve_results_for_enriched_event(self, job_id: str = None):
+        response = self.call_api(
+            "GET",
+            f"{self.host}/api/investigate/v2/orgs/{self.org_key}/enriched_events/search_jobs/{job_id}/results",
+            json_data={"job_id": job_id},
+        )
+
+        return response
+
+    def get_job_id_for_detail_search(self, event_id: str) -> Optional[str]:
+        response = self.call_api(
+            "POST",
+            f"{self.host}/api/investigate/v2/orgs/{self.org_key}/enriched_events/detail_jobs",
+            json_data={"event_ids": [event_id]},
+        ).get("job_id")
+
+        if response:
+            job_id = response
+            return job_id
+        return None
```

Comment:
```suggestion
        return response
```
This can be simplified, as it's the same as the 4 lines highlighted here